lib/src/easy_localization_app.dart

```diff
@@ -1,263 +1,149 @@
-        delegate = _EasyLocalizationDelegate(
-            path: path,
-            supportedLocales: supportedLocales,
-            useOnlyLangCode: useOnlyLangCode,
-            assetLoader: assetLoader),
-        super(key: key);
+    this.child,
+  }) : super(key: key);
 
-  _EasyLocalizationState createState() => _EasyLocalizationState();
+  static EasyLocalizationProvider of(BuildContext context) =>
+      EasyLocalizationProvider.of(context);
 
-  static _EasyLocalizationState of(BuildContext context) =>
-      _EasyLocalizationProvider.of(context).data;
+  @override
+  _EasyLocalizationState createState() => _EasyLocalizationState();
 }
 
-class _EasyLocalizationLocale extends ChangeNotifier {
-  Locale _locale;
-  static Locale _savedLocale;
-  static Locale _osLocale;
-  bool saveLocale;
-  VoidCallback onLocaleChange;
-
-  // @TOGO maybe add assertion to ensure that ensureInitialized has been called and that
-  // _savedLocale is set.
-  _EasyLocalizationLocale(
-      Locale fallbackLocale, List<Locale> supportedLocales, bool saveLocale, VoidCallback onLocaleChange)
-      : this.saveLocale = saveLocale,
-        this.onLocaleChange = onLocaleChange {
-    _init(fallbackLocale, supportedLocales);
+class _EasyLocalizationState extends State<EasyLocalization> {
+  Locale locale;
+  EasyLocalizationDelegate delegate;
+
+  _onLocaleChanged(Locale locale, [bool rebuild = true]) {
+    if (rebuild)
+      setState(() {
+        this.locale = locale;
+      });
+    else
+      this.locale = locale;
   }
 
-  //Initialize _EasyLocalizationLocale
-  _init(Locale fallbackLocale, List<Locale> supportedLocales){
-    // If saved locale then get
-    if (_savedLocale != null && this.saveLocale) {
-      log('easy localization: Saved locale loaded ${_savedLocale.toString()}');
-      locale = _savedLocale;      
-    } else {
-      locale = supportedLocales.firstWhere((locale) => _checkInitLocale(locale, _osLocale),
-          orElse: () => _getFallbackLocale(supportedLocales, fallbackLocale));
-    }
-    //Set locale
-    if (Intl.defaultLocale == null) locale = _locale;    
+  @override
+  void initState() {
+    loadSavedLocale();
+    delegate = EasyLocalizationDelegate(
+        path: widget.path,
+        supportedLocales: widget.supportedLocales,
+        useOnlyLangCode: widget.useOnlyLangCode,
+        assetLoader: widget.assetLoader,
+        onLocaleChanged: (locale) {
+          _onLocaleChanged(locale, false);
+        });
   }
 
-  bool _checkInitLocale(Locale locale, Locale osLocale) {
-    // If suported locale not contain countryCode then check only languageCode
-    if (locale.countryCode == null) {
-      return (locale.languageCode == _osLocale.languageCode);      
-    } else {
-      return (locale == _osLocale);
+  loadSavedLocale() async {
+    SharedPreferences _preferences = await SharedPreferences.getInstance();
+    var _codeLang = _preferences.getString('codeLa');
```

Comment:
I find bug, in new version @aissat use different sharedPreference val(locale), look _saveLocale function --> sharedPreference

Reply:
Good catch, I broke the test for taht. Sorry

Reply:
I will fix this later tonight, with the share preference rename the language loading broken. Ill push something later.

Reply:
don't worry about it

Reply:
Fixed!